### ISSUE TITLE
Plaintext Message "body" content

### ISSUE THREAD

**Issue by ashcherbakov** [decentralized-identity/didcomm-messaging#266]
[Plaintext Message Headers](https://identity.foundation/didcomm-messaging/spec/#message-headers) sections says that
> A DIDComm plaintext message conveys most of its application-level data inside a **JSON** body **object** that is a direct child of the message root. The structure inside body is predicted by the value of the message’s type attribute, and varies according to the definition of the protocol-specific message in question. Each type of message will have its own body.

It's not clear from that description and the provided examples what can be a "body" value:
- JSON Object only (basically key-value pairs), as it says `JSON Object` (see bold above).
- JSON Object or JSON Array
- any valid JSON (JSON object, or JSON array, or JSON value such as a number, string, bool)

In other words, what of the examples below are correct (or all examples are valid):

**Example 1**
```
{
  "type": "https://didcomm.org/my-protocol/1.0",
  "body": {
    "aaa": "bbb"
  }
  ...
}
```

**Example 2**
```
{
  "type": "https://didcomm.org/my-protocol/1.0",
  "body": "aaa"
  ...
}
```

**Example 3**
```
{
  "type": "https://didcomm.org/my-protocol/1.0",
  "body": 42
  ...
}
```

**Example 4**
```
{
  "type": "https://didcomm.org/my-protocol/1.0",
  "body": [
   {"aaa": "bbb"}, {"ccc": "ddd"}
  ]
  ...
}
```

**Comment by kdenhartog** [comment]
Doing a quick re-read on the JWM spec, it specifies: `The "body" attribute value is a JSON object conforming to RFC 7159 [RFC7159].`

Looking at [RFC7159 section 4](https://datatracker.ietf.org/doc/html/rfc7159#section-4) it states the following:

```
An object structure is represented as a pair of curly brackets
   surrounding zero or more name/value pairs (or members).  A name is a
   string.  A single colon comes after each name, separating the name
   from the value.  A single comma separates a value from a following
   name.  The names within an object SHOULD be unique.
```

So, I would think only Example 1 would be compatible. It may be worth referencing RFC7159 directly to make this a bit more clear in this spec I suppose so that readers don't have to dig through multiple specs to find the answer.
